consensus/src/epoch_manager.rs

```diff
@@ -1435,6 +1434,73 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
                 self.config.quorum_store.batch_expiry_gap_when_init_usecs;
             let payload_manager = self.payload_manager.clone();
             let pending_blocks = self.pending_blocks.clone();
+            let malicious_sender = self.epoch_state().verifier.is_malicious_author(&peer_id);
+
+            if self.config.optimistic_sig_verification_for_order_votes && !malicious_sender {
+                if let UnverifiedEvent::OrderVoteMsg(order_vote) = &unverified_event {
+                    order_vote.partial_verify()?;
+                    Self::forward_event_to(
+                        round_manager_tx,
+                        (
+                            peer_id,
+                            discriminant(&VerifiedEvent::UnverifiedOrderVoteMsg(
+                                order_vote.clone(),
+                            )),
+                        ),
+                        (
+                            peer_id,
+                            VerifiedEvent::UnverifiedOrderVoteMsg(order_vote.clone()),
+                        ),
+                    )
+                    .context("round manager sending unverified order vote to round manager")?;
+                    return Ok(());
+                }
+            }
+
+            if self.config.optimistic_sig_verification_for_votes && !malicious_sender {
+                if let UnverifiedEvent::VoteMsg(vote) = unverified_event.clone() {
+                    self.bounded_executor
+                        .spawn(async move {
+                            // The partial_verify function will potentially verify the signature of timeout.
```

Comment:
Actually, since timeout aggregation does not benefit much from optimistic sig and is in the unhappy path anyway, we don't need optimistic sig for timeout. So after Balaji's change, vote's optimistic sig does not need a separate thread.